Providing Equinox based REST Server and product deployment

PhilMFischer opened the pull request:
Based on #363 . Needs to be closed before.
Closes #356.

- Added a standalone RCP application for the Jetty Server
- Created Launch configuration for it
- Added product definition
- Updated deployment for server application

Commit:
Fixed dependency in funcelectrical ui plugin - (Task #361)

- Removed direct dependency to team.svn

---
Task #361: Make the VirSat repository available via REST interface.

Commit:
Write a Heading here - (Task #356)

- Added headless RCP application
- Restructured packages of server plugin
- Registered new application to plugin
- Added launch configuration for server

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Adding server feature to p2 updatesite - (Task #356)



---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Defined product for deployment - (Task #356)

- Added product definition for Virtual Satellite Server
- Updated poms for deploying new Server product

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

franzTobiasDLR commented on de.dlr.sc.virsat.server.feature/feature.xml:
```diff
@@ -0,0 +1,43 @@
+      plugin="de.dlr.sc.virsat.branding.ui"
+      license-feature="de.dlr.sc.virsat.license.feature">
+
+   <description>
```
Could we maybe update this pasted description?

PhilMFischer replied:
Yes its updated.

franzTobiasDLR commented on de.dlr.sc.virsat.external.lib.jersey/pom.xml:
```diff
@@ -0,0 +1,25 @@
+	<packaging>eclipse-plugin</packaging>
+
+
+</project>
```
I think not having file endings somethimes couses problems on linux... Could we fix that?

PhilMFischer replied:
Not sure about linux, but i don't mind adding it.

franzTobiasDLR commented on de.dlr.sc.virsat.external.lib.jersey/build.properties:
```diff
@@ -0,0 +1,37 @@
+               externalLib/jaxrs-ri/lib/jersey-media-json-binding.jar,\
+               externalLib/jaxrs-ri/lib/jersey-media-sse.jar,\
+               externalLib/jaxrs-ri/lib/jersey-server.jar
+	
```
Could we add a new line at the end of the file?

PhilMFischer replied:
well, there is a new line already. What exactly do you mean?

PhilMFischer replied:
Looking to the other comments i guess its solved

franzTobiasDLR commented on de.dlr.sc.virsat.external.lib.jersey/src/de/dlr/virsat/external/lib/jersey/Activator.java:
```diff
@@ -0,0 +1,21 @@
+public class Activator extends LibPlugin {
+
+	
+}
```
See above

PhilMFischer replied:
done

franzTobiasDLR commented on de.dlr.sc.virsat.server.test/pom.xml:
```diff
@@ -0,0 +1,23 @@
+	<artifactId>de.dlr.sc.virsat.server.test</artifactId>
+	<name>de.dlr.sc.virsat.server.test</name>
+	<packaging>eclipse-test-plugin</packaging>
+</project>
```
line ending again

PhilMFischer replied:
Done

franzTobiasDLR reviewed (approved):
Cool Stuff 👍 

There are some missing EOLs at the end of some files and the feature description of the server feature is copied from the team feature... Not really a show blocker for me but if you're working on it anyway you might want to fix it.

Commit:
Merging with latest from #359 - (Task #356)


---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Merge branch 'feature/task_359-Add_JGit_dependencies' into feature/task_356-Jetty_baseline_server

Commit:
Fixes due to review of @franzTobiasDLR - (Task #356)


---
Task #356: Jetty baseline server based on Virtual Satellite APPs

dianapeters-dlr commented on de.dlr.sc.virsat.external.lib.jersey/src/de/dlr/virsat/external/lib/jersey/servlet/ApplicationServletContainer.java:
```diff
@@ -0,0 +1,70 @@
+	}
+
+	/**
+	 * Implement this method to create the actual servlet with the correct
```
Somehow, the javadoc ends mid-sentence. Fix?

PhilMFischer replied:
Corrected.

dianapeters-dlr commented on de.dlr.sc.virsat.server.feature/feature.xml:
```diff
@@ -0,0 +1,55 @@
+   </description>
+
+   <copyright>
+      Copyright 2015
```
Shouldn't this be a current date? (If it is on purpose, then leave it - just wondering.)

PhilMFischer replied:
Same as above.

dianapeters-dlr commented on de.dlr.sc.virsat.server.test/about.html:
```diff
@@ -0,0 +1,27 @@
+<body lang="EN-US">
+<h2>About This Content</h2>
+ 
+<p>December 18, 2018</p>	
```
again: current date?

PhilMFischer replied:
Realy not sure :confused: . Need to be discussed in the team.

dianapeters-dlr commented on de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/data/VirSatGitAccessTest.java:
```diff
@@ -0,0 +1,157 @@
+	}
+
+	@Test
+	public void testCloneRepository() {
```
I would test only one case per method, because it is then easier to spot what exactly broke when single tests start failing. It is for example much more likely that the error message changes for which the negative case checks than that something fundametnal breaks - and it is easier to spot if the positive case still works.
(But that's personal preference.)

PhilMFischer replied:
It is probably a good idea. would be great if we could create an example together. I leave this one as it is for the moment.

Commit:
Refactored VirSat style GitAccess - (Task #356)

- Added a workspace folder to interface, and only allow too use relative
path for the repos.
- This is in preparation for a security manager and also for checking
out user specific repos.
- Updated and improved test case.


---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Merging with latest from #359 - (Task #356)

- Fixing test cases

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Started to implement User Context - (Task #356)

- Added user context for locking
- Added test cases.

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Merged - (Task #356)

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Merged some changes - (Task #356)

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Introducing User Context - (Task #356)

- Adding User Context to model access resource
- Adjusted test case to user context change

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Introducing User Context - (Task #356)

- Adding User Context to model access resource
- Adjusted test case to user context change

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Fixing issue with relative path for tests - (Task #356)

- CHanging to a pure relative path, otherwise the test fails on the
linux based CI environment.

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Fixing feature.xml - (Task #356)

- Fixed damage from merge
- Fixed version dependency in tests feature

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Fixes due to latest review - (Task #356)

- Fixed some javadoc comment.

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

SaMuellerDLR commented on de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/WorkspaceUserContext.java:
```diff
@@ -0,0 +1,102 @@
+	/**
+	 * Call this method to execute code within the context
+	 * @param workspaceExecution AN implementation to be called within the context
+	 * @return A STring as the result of the code executed in the context
```
"STring" -> "String"

PhilMFischer commented on de.dlr.sc.virsat.external.lib.jersey/src/de/dlr/virsat/external/lib/jersey/servlet/ApplicationServletContainer.java:
```diff
@@ -0,0 +1,69 @@
+
+	@Override
+	public void init(ServletConfig arg0) throws ServletException {
+		this.servlet.init(arg0);
```
Agree

PhilMFischer commented on de.dlr.sc.virsat.external.lib.jersey/src/de/dlr/virsat/external/lib/jersey/servlet/ApplicationServletContainer.java:
```diff
@@ -0,0 +1,69 @@
+
+	@Override
+	public void service(ServletRequest arg0, ServletResponse arg1) throws ServletException, IOException {
+		this.servlet.service(arg0, arg1);
```
Agree

Commit:
Requested Fixes by Review - (Task #356)

- Gave better parameter names in methods of ServletContainer.
- Fixed STring to String in javaDoc statement.

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

PhilMFischer commented:
Just an initial commit of server code. Will be refactored anyway. This is basically just the prototype without much of functinality.

Commit:
Merge branch 'development' into feature/task_356-Jetty_baseline_server

Conflicts:
	de.dlr.sc.virsat.server.feature/feature.xml
	de.dlr.sc.virsat.server/META-INF/MANIFEST.MF
	de.dlr.sc.virsat.test.feature/feature.xml

Commit:
Merge remote-tracking branch 'origin/development' into feature/task_356-Jetty_baseline_server

SaMuellerDLR reviewed (approved):
Approved!
